domain/ide/workspace/folder.go

```diff
@@ -226,7 +213,45 @@ func (f *Folder) processResults(scanData snyk.ScanData) {
 	f.FilterAndPublishCachedDiagnostics(scanData.Product)
 }
 
+func incrementSeverityCount(scanData *snyk.ScanData, issue snyk.Issue) {
+	issueProduct := issue.Product
+	if issueProduct == "" {
+		log.Debug().Str("method", "incrementSeverityCount").Msg("Issue product is empty. Setting to unknown")
+		issueProduct = "unknown"
+	}
+
+	initializeSeverityCountForProduct(scanData, issueProduct)
+
+	switch issue.Severity {
+	case snyk.Critical:
+		scanData.SeverityCount[issueProduct].Critical++
+	case snyk.High:
+		scanData.SeverityCount[issueProduct].High++
+	case snyk.Medium:
+		scanData.SeverityCount[issueProduct].Medium++
+	case snyk.Low:
+		scanData.SeverityCount[issueProduct].Low++
+	}
+}
+
+func initializeSeverityCountForProduct(scanData *snyk.ScanData, productType product.Product) {
+	if scanData.SeverityCount == nil {
```

Comment:
question: why is SeverityCount a pointer? Not a big issue, but not using a pointer - if possible - may make it easier to use & understand.

Reply:
In the beginning of this refactor, I decided to use it as a pointer because I didn't know how to assert it in the test. Then you taught me that I could initialise the map in the mock data here: 

```go
	scanData := snyk.ScanData{
		Product:       product.ProductOpenSource,
		SeverityCount: make(map[product.Product]snyk.SeverityCount),
		Issues:        []snyk.Issue{issue},
	}
```

Now after your question, I removed the pointer and I'm committing the changes.